components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/DBPersistenceStore.java

```diff
@@ -0,0 +1,337 @@
+                if (databaseType.equals(MSSQL_DATABASE_TYPE)) {
+                    blobSnapshot = new SerialBlob(resultSet.getBytes("snapshot"));
+                } else {
+                    blobSnapshot = resultSet.getBlob("snapshot");
+                }
+                int blobLength = (int) blobSnapshot.length();
+                blobAsBytes = blobSnapshot.getBytes(1, blobLength);
+            }
+
+        } catch (SQLException e) {
+            log.error("Error while retrieving revision " + revision + " of siddhiApp: " +
+                    siddhiAppName + " from the database with datasource " + datasourceName, e);
+        } finally {
+            cleanupConnections(stmt, con);
+        }
+        return blobAsBytes;
+    }
+
+    @Override
+    public String getLastRevision(String siddhiAppName) {
+        createTableIfNotExist();
+        PreparedStatement stmt = null;
+        Connection con = null;
+        String revision = null;
+        try {
+            try {
+                con = datasource.getConnection();
+            } catch (SQLException e) {
+                log.error("Cannot establish connection to datasource " + datasourceName +
+                        " while trying retrieve last revision of " + siddhiAppName, e);
+                return null;
+            }
+            stmt = con.prepareStatement(executionInfo.getPreparedSelectLastStatement());
+            stmt.setString(1, siddhiAppName);
+            ResultSet resultSet = stmt.executeQuery();
+            if (resultSet.next()) {
+                revision = String.valueOf(resultSet.getString("revision"));
+            }
+        } catch (SQLException e) {
+            log.error("Error while retrieving last revision of siddhiApp: " +
+                    siddhiAppName + "from the database with datasource " + datasourceName, e);
+        } finally {
+            cleanupConnections(stmt, con);
+        }
+
+        return revision;
+    }
+
+    private void initializeDatabaseExecutionInfo() {
+        executionInfo = new ExecutionInfo();
+        RDBMSQueryConfigurationEntry databaseQueryEntries =
+                RDBMSConfiguration.getInstance().getDatabaseQueryEntries(databaseType, tableName);
+
+        try {
+            executionInfo.setPreparedInsertStatement(databaseQueryEntries.getInsertTableQuery());
+            executionInfo.setPreparedCreateTableStatement(databaseQueryEntries.getCreateTableQuery());
+            executionInfo.setPreparedTableExistenceCheckStatement(databaseQueryEntries.getIsTableExistQuery());
+            executionInfo.setPreparedSelectStatement(databaseQueryEntries.getSelectTableQuery());
+            executionInfo.setPreparedSelectLastStatement(databaseQueryEntries.getSelectLastQuery());
+            executionInfo.setPreparedDeleteStatement(databaseQueryEntries.getDeleteQuery());
+            executionInfo.setPreparedCountStatement(databaseQueryEntries.getCountQuery());
+        } catch (NullPointerException e) {
```

Comment:
databaseQueryEntries might become null if the database type is not supported and thrown an NPE